contracts/lib/DexLibrary.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "./SafeMath.sol";
+import "../interfaces/IPair.sol";
+import "../interfaces/IWAVAX.sol";
+
+library DexLibrary {
+    using SafeMath for uint;
+    bytes private constant zeroBytes = new bytes(0);
+    IWAVAX private constant WAVAX = IWAVAX(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7);
+
+
+    /**
+     * @notice Swap directly through a Pair
+     * @param amountIn input amount
+     * @param fromToken address
+     * @param toToken address
+     * @param pair Pair used for swap
+     * @return output amount
+     */
+    function swap(uint amountIn, address fromToken, address toToken, IPair pair) internal returns (uint) {
+        (address token0,) = sortTokens(fromToken, toToken);
+        (uint112 reserve0, uint112 reserve1,) = pair.getReserves();
+        if (token0 != fromToken) (reserve0, reserve1) = (reserve1, reserve0);
+        uint amountOut1 = 0;
+        uint amountOut2 = getAmountOut(amountIn, reserve0, reserve1);
+        if (token0 != fromToken) (amountOut1, amountOut2) = (amountOut2, amountOut1);
+        safeTransfer(fromToken, address(pair), amountIn);
+        pair.swap(amountOut1, amountOut2, address(this), zeroBytes);
+        return amountOut2 > amountOut1 ? amountOut2 : amountOut1;
+    }
+
+    function checkSwapPairCompatibility(IPair pair, address tokenA, address tokenB) internal pure returns (bool) {
+        return (tokenA == pair.token0() || tokenA == pair.token1()) && (tokenB == pair.token0() || tokenB == pair.token1()) && tokenA != tokenB;
+    }
+
+    function estimateConversionThroughPair(uint amountIn, address fromToken, address toToken, IPair swapPair) internal view returns (uint) {
+        (address token0,) = sortTokens(fromToken, toToken);
+        (uint112 reserve0, uint112 reserve1,) = swapPair.getReserves();
+        if (token0 != fromToken) (reserve0, reserve1) = (reserve1, reserve0);
+        return getAmountOut(amountIn, reserve0, reserve1);
+    }
+
+    /**
+     * @notice Converts reward tokens to deposit tokens
+     * @dev No price checks enforced
+     * @param amount reward tokens
+     * @return deposit tokens
+     */
+    function convertWAVAXToDepositTokens(uint amount, address depositToken, IPair swapPairToken0, IPair swapPairToken1) internal returns (uint, uint) {
```

Comment:
Check `_convertRewardTokensToDepositTokens` from the DexV6, this function could be generalized like that one just passing an extra argument "ERC20 rewardToken"